### ISSUE TITLE
Setup using pip broken since CairoSVG dropped Python 2 support

### ISSUE THREAD

**Issue by KarelJakubec** [Kozea/WeasyPrint#394]
Hi Kozea,
I am experiencing problems with installing WeasyPrint 0.32 into my python 2.7.12 virtualenv using pip. It is caused by new release of CairoSVG (2.0), which setup fails cause it no longer supports Python 2.

Problem seems to be in METADATA file which contains dependency on cairosvg >= 1.0.20. This automatically downloads CairoSVG 2.0 and so it fails.

Metadata-Version: 2.0
Name: WeasyPrint
Version: 0.32
Summary: WeasyPrint converts web documents to PDF.
Home-page: http://weasyprint.org/
Author: Simon Sapin
Author-email: simon.sapin@kozea.fr
License: BSD
Platform: UNKNOWN
Classifier: Development Status :: 5 - Production/Stable
Classifier: Intended Audience :: Developers
Classifier: License :: OSI Approved :: BSD License
Classifier: Programming Language :: Python :: 2
Classifier: Programming Language :: Python :: 2.7
Classifier: Programming Language :: Python :: 3
Classifier: Programming Language :: Python :: 3.3
Classifier: Programming Language :: Python :: 3.4
Classifier: Programming Language :: Python :: 3.5
Classifier: Topic :: Internet :: WWW/HTTP
Classifier: Topic :: Text Processing :: Markup :: HTML
Classifier: Topic :: Multimedia :: Graphics :: Graphics Conversion
Classifier: Topic :: Printing
Requires-Dist: CairoSVG (>=1.0.20)
Requires-Dist: Pyphen (>=0.8)
Requires-Dist: cairocffi (>=0.5)
Requires-Dist: cffi (>=0.6)
Requires-Dist: cssselect (>=0.6)
Requires-Dist: html5lib (>=0.999999999)
Requires-Dist: lxml (>=3.0)
Requires-Dist: tinycss (>=0.4)

I've seen that there is already a patch solving this problem in setup.py, which should limit the CairoSVG version to 1.0.20 but somehow it is not applied in package in PyPI. I would suggest that the package was built in Python 3 environment and so the extra dependency was therefore skipped. However I am not an expert on Python packaging so I am not sure if such a scenario is even possible.

The workaround to make things working is to set CairoSVG==1.0.20 in your requirements.txt file (or whenever you store packages your projects) which should effectively freeze it.

**Comment by liZe** [comment]
There's some code in `setup.py` to avoid this, but for some reason it doesn't work (maybe because of the wheel).

**Comment by liZe** [comment]
I've removed the wheel, it seems to work now. @KarelJakubec You can close this issue if it's fixed for you too.

**Comment by aschosser** [comment]
Hi Karel

> I am experiencing problems with installing WeasyPrint 0.32 into my python 2.7.12 virtualenv using pip. It is caused by new release of CairoSVG (2.0), which setup fails cause it no longer supports Python 2.

Which version are you updating from? I experienced the same problem
and solve it by upgrading step by step, from 0.29 to 0.30 and so on.

Andreas

-- 
Kurs 10 IT-Consulting                           www.kurs-10.de
Andreas Schosser                                  a@kurs-10.de

Baldestraße. 14                      Telefon +49 89 41615842-0
80469 München                        Telefax +49 89 41615842-3

0x6EDECCF1 - 2AA0 939B 5585 819B FCE8 E43B 0B8E 0DF2 6EDE CCF1

**Comment by KarelJakubec** [comment]
Hi,
@aschosser it was not an upgrade but a fresh installation. We have a machine setup which had  WeasyPrint freezed to 0.26. Once CairoSVG 2.0 was released it became to fail. At first I though it would be enough just to update to the newest WeasyPrint, but I experienced the same problem, so I examined the dependencies and the rest is described above...:)

@liZe Anyway it works for me now, so I am closing the issue.

Thank you for help, best regards
Karel
